### PR TITLE
add lookup_singular_library_symbol_wo_rng

### PR DESCRIPTION
**Pull request by tthsqe12**

### CODE REVIEW

File: caller.cpp [hunk 1]
```diff
@@ -298,7 +298,26 @@ jl_value_t * call_singular_library_procedure_wo_rng(
...
+    int progress = 0;
+    leftv u = (leftv) IDROOT->get(pack.c_str(),0);
+    if (u != NULL)
+    {
+        progress++;
+        idhdl v = ((package)(u->Data()))->idroot->get(name.c_str(), 0);
+        if (v != NULL)
+        {
+            sleftv x;
+            x.Init();
```
Comment (hannes14) [inline caller.cpp:314]: this x.Init() is not needed as the follwing x.Copy includes Init()

File: caller.cpp [hunk 2]
```diff
@@ -298,7 +298,26 @@ jl_value_t * call_singular_library_procedure_wo_rng(
...
+            x.Copy((leftv)v);
+            return get_julia_type_from_sleftv(&x);  // any cleanup for x ???
```
Comment (hannes14) [inline caller.cpp:316]: no cleanup needed: x contains (here) only the type (int) and the raw pointer with gets reused in get_julia_type_from_sleftv

**Review by hannes14 — APPROVED** [review]

File: caller.cpp [hunk 3]
```diff
@@ -298,7 +298,25 @@ jl_value_t * call_singular_library_procedure_wo_rng(
...
+    if (u != NULL)
+    {
+        progress++;
+        idhdl v = ((package)(u->Data()))->idroot->get(name.c_str(), 0);
+        if (v != NULL)
+        {
+            sleftv x;
+            x.Copy((leftv)v);
+            return get_julia_type_from_sleftv(&x);
+        }
```
Comment (hannes14) [inline caller.cpp:316]: else progress=0;
to catch the case that pack is found, but name not (i.e. v==NULL)

**Review by hannes14 — APPROVED** [review]

File: caller.cpp (hunk 3 above)
Comment (tthsqe12) [reply caller.cpp:316]: I use progress=0 for package not found, and progres=1 for package found but symbol not found. The error message will distinguish these cases.

**Pull request merged**